WarshipGame/Assets/Scripts/Gameplay/Camera/CameraController.cs

```diff
@@ -0,0 +1,175 @@
+        move = Quaternion.Euler(0,-90,0) * move;
+        Vector3 camPosMove = new Vector3(_cameraOrigin.x + move.x * Time.deltaTime * PanningSpeed, _cameraOrigin.y, _cameraOrigin.z + move.z * Time.deltaTime * PanningSpeed);
+
+        if (camPosMove.z <= -MapSizeZ || camPosMove.z >= MapSizeZ || camPosMove.x <= -MapSizeX || camPosMove.x >= MapSizeX)
+        {  
+            return;
+        }
+        
+        CMVirtualCamera.transform.position =  camPosMove;
+    }
+
+    /// <summary>
+    /// Corrects the screen panning by correcting the screen size and FOV distance
+    /// relative to the camera width and height
+    /// </summary>
+    /// <returns></returns>
+    private Vector2 ScaleScreenToWorldSize(float camAspect, float camFOV, float camScreenPixelWidth,
+        float camScreenPixelHeight, float screenW, float screenH)
+    {
+        float cameraWidth = camAspect * camFOV * 2f;
+        float cameraHeight = camFOV * 2f;
+        float screenWorldW = screenW * (cameraWidth / camScreenPixelWidth);
+        float screenWorldH = screenH * (cameraHeight / camScreenPixelHeight);
+        
+        return new Vector2(screenWorldW, screenWorldH);
+    }
+
+    /// <summary>
+    /// Zooms the camera when the scrolling axis is changed
+    /// </summary>
+    private void ZoomCamera()
+    {
+        float scrollValue = Input.GetAxis("Mouse ScrollWheel");
+        float newZoomValue = _currentFOVValue -= scrollValue * ZoomSpeed;
+
+        if (newZoomValue <= MaxZoom.x)
+        {
+            _currentFOVValue = MaxZoom.x;
+            CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+            return;
+        }
+        
+        if (newZoomValue >= MaxZoom.y)
+        {
+            _currentFOVValue = MaxZoom.y;
+            CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+            return;
+        }
+        
+        _currentFOVValue = newZoomValue;
+        CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+    }
+
+    /// <summary>
+    /// A function that should be added on the OnShipSelected Unity Event to invoke a focus
+    /// This Function moves to a selected ship and changes to a close-up FOV
+    /// </summary>
+    public void Focus(GameObject result)
+    {
+        _isFocussing = true;
+        Vector3 vCamPosition = transform.position;
+        
+        Physics.Raycast(vCamPosition,transform.forward,  out RaycastHit hit);
+        
+        Vector3 resultPosition = result.transform.position;
+        Vector3 rayHitPosition = hit.point;
+        
+        
+        Vector2 selectedShipPosition = new Vector2(resultPosition.x, resultPosition.z);
+        Vector2 hitPosition = new Vector2(rayHitPosition.x, rayHitPosition.z);
+        
+        Vector2 difference = hitPosition - selectedShipPosition;
+
+        Vector3 newPosition = new Vector3(vCamPosition.x - difference.x, vCamPosition.y, vCamPosition.z - difference.y);
```

Comment:
Because it is already defined as a Vector3 and Vector2 we don't need to say we are making a new Vector3 and Vector2 just new